sentry-opentelemetry/sentry-opentelemetry-core/src/test/kotlin/SentrySpanProcessorTest.kt

```diff
@@ -85,64 +114,336 @@ class SentrySpanProcessorTest {
 
     @Test
     fun `ignores sentry client request`() {
-        val tracer = fixture.getSut()
-        tracer.spanBuilder("testspan")
-            .setSpanKind(SpanKind.CLIENT)
+        fixture.setup()
+        givenSpanBuilder(SpanKind.CLIENT)
             .setAttribute(SemanticAttributes.HTTP_URL, "https://key@sentry.io/proj/some-api")
             .startSpan()
 
-        verify(fixture.hub, never()).startTransaction(any<TransactionContext>(), any<TransactionOptions>())
+        thenNoTransactionIsStarted()
     }
 
     @Test
     fun `ignores sentry internal request`() {
-        val tracer = fixture.getSut()
-        tracer.spanBuilder("testspan")
-            .setSpanKind(SpanKind.CLIENT)
+        fixture.setup()
+        givenSpanBuilder(SpanKind.CLIENT)
             .setAttribute(SemanticAttributes.HTTP_URL, "https://key@sentry.io/proj/some-api")
             .startSpan()
 
-        verify(fixture.hub, never()).startTransaction(any<TransactionContext>(), any<TransactionOptions>())
+        thenNoTransactionIsStarted()
+    }
+
+    @Test
+    fun `does nothing on start if Sentry has not been initialized`() {
+        val hub = mock<IHub>()
```

Comment:
s: Any reason why you're not using the fixture's hub here?

Reply:
Just to make sure it's a clean mock that doesn't respond to anything but the wanted invocation on the hub. Rest is mocks that differ from other tests anyways. Guess it's also easier to read with the fresh mock instead of getting the one from the fixture that makes people read through all the setup code.